### DISCUSSION TITLE
Any plans to support CBOR sequences?

### DISCUSSION THREAD

**Discussion by gurki** [nlohmann/json#4432] [Q&A]
We use `.jsonl` / `.cborseq` commonly for logging and capturing data streams over time. While JSON Lines is not a standardized format (yet afaik, https://jsonlines.org/), CBOR Sequence is (https://www.rfc-editor.org/rfc/rfc8742.html). 
I was looking through the issues and discussions, but couldn't find any mention of it. Are there plans to add support for them? If not, and if it's not out of the scope of nlohmann - and I think both would make a fantastic addition - would be happy to take a look and propose an implementation.

**Comment by nlohmann** [comment]
There is currently no built-in support for RFC 8742 CBOR sequences (or similar .cborseq multi-value binary streams). Writing a sequence is already trivial -- repeated to_cbor() calls into the same output stream/adapter just concatenate valid CBOR items back to back -- but reading one back is not well supported: from_cbor(first, last, strict=false) stops after the first value but does not expose how many bytes were consumed, because the iterator is moved-from into the input adapter with no public way to recover the end position. That's the same missing building block asked about in discussion #5117 ('track bytes consumed by from_cbor'); a PR adding a from_cbor variant/overload that reports consumed bytes would be the natural first step and would likely be welcome, per the usual contribution process.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
